core/finality-grandpa/src/communication/tests.rs

```diff
@@ -257,8 +303,19 @@ fn good_commit_leads_to_relay() {
 			// when the commit comes in, we'll tell the callback it was good.
 			let handle_commit = commits_in.into_future()
 				.map(|(item, _)| {
-					let (_, _, mut callback) = item.unwrap();
-					(callback)(super::CommitProcessingOutcome::Good);
+					match item.unwrap() {
+						grandpa::voter::CommunicationIn::Commit(_, _, mut callback) => {
+							// NOTE: the type is opaque and there's no way to
+							// create it outside of the `finality-grandpa`
+							// crate. currently we don't do anything with it so
+							// it's safe to keep it uninitialized
+							let data: grandpa::voter::GoodCommit = unsafe {
```

Comment:
the idea of potentially having UB in a test scares me

Reply:
Added a `test-helpers` feature to `finality-grandpa` to allow creating these types.